ArmyOfOnes/ArmyOfOnes/ViewControllers/CurrenciesListViewController.swift

```diff
@@ -0,0 +1,216 @@
+        
+        setQuantityInputLayout()
+        setupRates()
+        
+        view.setNeedsUpdateConstraints()
+    }
+    
+    func setQuantityInputLayout() -> Void {
+        
+        quantityInput.delegate = self
+        quantityInput.addTarget(self, action: #selector(CurrenciesListViewController.textFieldChanged(_:)), for: UIControlEvents.editingChanged)
+        navigationController?.view.insertSubview(quantityInput, belowSubview: navigationController!.navigationBar)
+        
+        //sets format number style to currency
+        currencyFormatter.numberStyle = NumberFormatter.Style.currency
+    }
+    
+    func showNoConnectionMessage() -> Void {
+        //shows an alert message when there is no internet connection
+        let alert = UIAlertController(title: "Connection error", message: "Make sure you are connected to the Internet", preferredStyle: UIAlertControllerStyle.alert)
+        alert.addAction(UIAlertAction(title: "OK", style: UIAlertActionStyle.default, handler: nil))
+        present(alert, animated: true, completion: nil)
+    }
+    
+    func setupRates() -> Void {
+        
+        let currencyController = CurrencyController()
+        currencyController.fetchRates(Country.allCountries) {(fetchedRates, error) -> Void in
+            
+            
+            guard let _ = error else  {
+                
+                guard let ratesList = fetchedRates else {
+                    print("rates list is not defined")
+                    return
+                }
+                
+                self.rates = ratesList
+                self.reloadRates()
+                
+                return
+            }
+            
+            self.showNoConnectionMessage()
+            
+        }
+
+    }
+    
+    func reloadRates() -> Void {
+        currencyRatesList.removeAll()
+        currencyCountries.removeAll()
+        
+        for rate in rates {
+            let countryValue = rate.country.rawValue
+            let rateValue = rate.value * dollarQuantity
+            currencyFormatter.currencyCode = countryValue
+            let convertedRateValue = NSNumber(value: rateValue)
+
+            if let currencyFormatedValue = currencyFormatter.string(from: convertedRateValue) {
+                currencyRatesList.append("\(countryValue) \(currencyFormatedValue)")
+                currencyCountries.append("\(countryValue)")
+            }
+            DispatchQueue.main.async{
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    //MARK: Snapkit methods
+    
+    override func updateViewConstraints() {
+        
+        if (!didSetupConstraints) {
+            
+            if let superview = view {
+                
+                let inputOffset = LayoutConstraints.inputOffset.rawValue
+                let topInset = LayoutConstraints.topInset.rawValue
+                
+                quantityInput.snp.makeConstraints { make in
```

Comment:
You can set the constraints when the view controller is loaded on the screen. That way you don't need to have a state to know whether the constraints were already set